File: chroma-league/src/test/java/com/bonepl/chromaleague/hud/animations/LoadingAnimationTest.java
package com.bonepl.chromaleague.hud.animations;

import com.bonepl.chromaleague.hud.AnimationTester;
import org.junit.jupiter.api.Test;

class LoadingAnimationTest {
    @Test
    void testLoadingAnimation() {
        final LoadingAnimation loadingAnimation = new LoadingAnimation();
        new AnimationTester()
                .testAnimation(loadingAnimation, 100);
    }
}

File: test/hud/animations/LoadingAnimation.test.ts
import {describe, test} from 'vitest';
import {AnimationTester} from '../../AnimationTester.js';
import {LoadingAnimation} from '../../../src/hud/animations/LoadingAnimation.js';

/**
 * Integration test for loading animation on actual hardware.
 */
describe('LoadingAnimation Integration', () => {
    test('testLoadingAnimation', async () => {
        const loadingAnimation = new LoadingAnimation();
        await new AnimationTester().testAnimation(loadingAnimation, 100);
    }, 30000);
});
